sys/dev/puc/puc_pci.c

```diff
@@ -198,3 +198,6 @@ static driver_t puc_pci_driver = {
 };
 
 DRIVER_MODULE(puc, pci, puc_pci_driver, puc_devclass, 0, 0);
+MODULE_PNP_INFO("U16:vendor;U16:device;U16:#;U16:#;D:#", pci, puc,
+		puc_pci_devices, sizeof(puc_pci_devices[0]),
+		nitems(puc_pci_devices) - 1);
```

Comment:
This doesn't compile because the size of puc_pci_devices isn't known here.

Reply:
I see, that might be because this table is not present directly in the same file. It is present in another one. What could be done here to correct this ?

Reply:
I think you'll need to rearrange the code a little so that puc_pci_devices is included in pcu_pci.c somehow. I'd recommend doing it with a #include of the current file that has it since I believe that file is shared with NetBSD.

Reply:
Included pucdata.c which contains puc_pci_devices

Reply:
missed the reply to this until now...

This looks like it could work, but pucdata.c is already built in the module and specified in the kernel config files (see sys/conf/files for lines containing pucdata.c). So, adjustments in those areas are needed:

diff --git a/sys/conf/files b/sys/conf/files
index d680535ae0b9..b2da16980cee 100644
--- a/sys/conf/files
+++ b/sys/conf/files
@@ -2771,7 +2771,6 @@ dev/puc/puc.c                     optional puc
 dev/puc/puc_cfg.c              optional puc
 dev/puc/puc_pccard.c           optional puc pccard
 dev/puc/puc_pci.c              optional puc pci
-dev/puc/pucdata.c              optional puc pci
 dev/quicc/quicc_core.c         optional quicc
 dev/ral/rt2560.c               optional ral
 dev/ral/rt2661.c               optional ral
diff --git a/sys/modules/puc/Makefile b/sys/modules/puc/Makefile
index 042ae102abef..882e33343067 100644
--- a/sys/modules/puc/Makefile
+++ b/sys/modules/puc/Makefile
@@ -4,7 +4,7 @@
 .PATH: ${SRCTOP}/sys/dev/puc

 KMOD=  puc
-SRCS=  puc.c puc_cfg.c puc_pci.c puc_pccard.c pucdata.c
+SRCS=  puc.c puc_cfg.c puc_pci.c puc_pccard.c
 SRCS+= bus_if.h device_if.h serdev_if.c serdev_if.h \
        card_if.h pci_if.h

If you do any future pull requests, please make sure they compile.